### PR TITLE
New fields moved from traits to main member index, and migration script

### PR DESCRIPTION
**Pull request by jmgasper**

### CODE REVIEW

File: src/scripts/migrate-traits.js [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+    responseQueue.push(response)
+    while (responseQueue.length) {
+        const body = responseQueue.shift()
+        console.log("Loading traits: " + searchResults.hits.hits.length + " of: " + body.hits.total)
+
+        // collect the titles from this response
+        body.hits.hits.forEach(function (hit) {
+            searchResults.hits.hits.push(hit)
+            //searchResults.push(hit._source.quote)
```
Comment (testflyjets) [inline src/scripts/migrate-traits.js:45]: If this line isn't needed let's remove it.

File: src/scripts/migrate-traits.js [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+        })
+
+        // check to see if we have collected all of the quotes
+    if (body.hits.total === searchResults.hits.hits.length) {
+      searchResults.hits.total=body.hits.total
```
Comment (testflyjets) [inline src/scripts/migrate-traits.js:50]: Add spaces on either side of the `=`

File: src/scripts/migrate-traits.js [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+  while (responseQueue.length) {
+    const body = responseQueue.shift()
+    console.log("Loading members: " + searchResults.hits.hits.length + " of: " + body.hits.total)
+    
+    // collect the titles from this response
+    body.hits.hits.forEach(function (hit) {
+      searchResults.hits.hits.push(hit)
+    })
+
+    // check to see if we have collected all of the quotes
```
Comment (testflyjets) [inline src/scripts/migrate-traits.js:91]: I realize this is a one-off script and I'm guessing this code came from a similar script dealing with "quotes" but can we modify the descriptions here to describe what we're actually operating on -- "titles" and "quotes" doesn't seem accurate.

File: src/scripts/migrate-traits.js [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+    }
+    
+    dbMember.skillScoreDeduction = skillScoreDeduction
+    
+    console.log(dbMember.handle, "Skill score deduction", skillScoreDeduction)
+    console.log(dbMember.handle, "Names and handle appearance", dbMember.namesAndHandleAppearance)
+    console.log(dbMember.handle, "Available for gigs", dbMember.availableForGigs)
+    if(applyForReal){
+        try{
+            const result = await helper.update(dbMember, {"skillScoreDeduction":skillScoreDeduction})
```
Comment (testflyjets) [inline src/scripts/migrate-traits.js:153]: Need a space here after the colon: `{"skillScoreDeduction": skillScoreDeduction}` 

Also, for JS code, do we not typically add spaces between curly braces and their contained contents? Other languages do, I just don't recall if we use that style here.

File: src/scripts/migrate-traits.js [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+  
+    getESData()
+    .then(result => {
+        getMemberTraits()
+            .then(traits => {
+                const memberTraits = _.map(traits, (item) => item._source)
+                const members = _.map(result, (item) => item._source)
+                const membersWithTraits = _.map(members, function (item) {
+                traits = memberTraits.filter( member => member.userId === item.userId)
+                if (traits && traits.length>0) {
```
Comment (testflyjets) [inline src/scripts/migrate-traits.js:195]: Spaces here: `traits.length > 0`

File: src/services/MemberService.js [hunk 6]
```diff
@@ -21,7 +21,7 @@ const lookerService = new LookerApi(logger)
 
 const MEMBER_FIELDS = ['userId', 'handle', 'handleLower', 'firstName', 'lastName', 'tracks', 'status',
   'addresses', 'description', 'email', 'homeCountryCode', 'competitionCountryCode', 'photoURL', 'verified', 'maxRating',
-  'createdAt', 'createdBy', 'updatedAt', 'updatedBy', 'loginCount', 'lastLoginDate', 'skills']
+  'createdAt', 'createdBy', 'updatedAt', 'updatedBy', 'loginCount', 'lastLoginDate', 'skills', 'availableForGigs', 'skillScoreDeduction', 'namesAndHandleAppearance']
```
Comment (testflyjets) [inline src/services/MemberService.js:24]: Maybe wrap this line since it's very long now?

**Review by testflyjets — APPROVED** [review]
A few formatting comments, but looks good otherwise.

File: src/scripts/migrate-traits.js (hunk 1 above)
Comment (jmgasper) [reply src/scripts/migrate-traits.js]: Done

File: src/scripts/migrate-traits.js (hunk 3 above)
Comment (jmgasper) [reply src/scripts/migrate-traits.js]: Done

File: src/services/MemberService.js (hunk 6 above)
Comment (jmgasper) [reply src/services/MemberService.js]: Done

**Pull request merged**